libs/blocks/marquee/marquee.js

```diff
@@ -136,4 +146,8 @@ export default async function init(el) {
   if (el.classList.contains('mnemonic-list') && foreground) {
     await loadMnemonicList(foreground);
   }
+
+  if (el.classList.contains('countdown-timer')) {
+    await loadCountDownTimer(el);
```

Comment:
I'm a little worried about this, does the countdown-timer have a fixed size? The best order of operations would be:
- Marquee has the countdown-timer class
- We have some CSS to reserve the height/width for it
- We render the marquee and load the countdown timer to add it to the DOM once it finished loading

And this order of operations should happen _non blocking_ currently we wait for rendering the marquee which will cause higher LCP times => aka this will eat into our performance budget

Reply:
Can we nudge designs to reserve a little more space than needed to avoid CLS? 🤔 

Two issues to unwrap:
- Fragments within the LCP = bad for performance
- We add 4 requests, 3 of them chained to the LCP

Before
![Screenshot 2024-09-02 at 15 55 54](https://github.com/user-attachments/assets/5969e21e-700c-4fee-8c31-e3ca257af1ee)

After
![Screenshot 2024-09-02 at 15 56 02](https://github.com/user-attachments/assets/402c6f47-52f1-4f44-8e60-a62a8c8d8162)



If we look at a very plain marquee, we have this more condensed timeline.
![Screenshot 2024-09-02 at 16 02 15](https://github.com/user-attachments/assets/9e2d3d88-e466-4afd-8608-d552e1fe070e)

Reply:
@mokimo why CLS? the section would be shown only when all the contained blocks are rendered.
Assuming dimensions of your content is not a good idea as it depends on the site locale.

Regarding the LCP,  in some cases marquee contains other blocks such as countdown or mnemonics. 
Depending on the outcome of the [performance discussion](https://github.com/orgs/adobecom/discussions/2719)  we can preload all marquee dependencies to avoid the prolonged waterfall.

Reply:
I think the `mnemonics` is already an anti pattern that wasn't caught early enough. We should not repeat the same mistake

Reply:
I guess prelinks could improve situation, but is it possible to automatically decide which resources are to be preloaded? Or will it be manually maintained? The second is not feasible for promos.

the other way I could see it implemented would be to decouple countdown content in some reusable location (promo manifest? or promotions.xlsx) and to have a consonant design of marquee, text, ... block including countdown. However, this solution is unlikely to be delivered before Black Friday authoring begins.

Reply:
We should work with consonant to produce a design to _reserve space_ and place this element consistently within the marquee so that we can avoid waiting for the element to download.

This has been a fair amount of work for the global navigation, but we also consistently reserve space for that on every layout, with breadcrumbs and without, mobile, tablet, completely authorable etc.

Reply:
@mokimo what you are mentioning is totally different than what has been advertised for Milo w.r.t section/block for CLS/LCP.  On Milo, you block the main thread to render but you are asked to keep it short so that you don't have to worry about CLS as what you render is made visible only after properly rendering in DOM.

>I think the mnemonics is already an anti pattern
@Axelcureno is on PTO but the feature was discussed at https://github.com/orgs/adobecom/discussions/1642.

If we manage to load nested blocks together with marquee, and not after the marquee we would be at a good position.

Currently `loadBlock` in Milo doesn't check for nested blocks. Maybe we can simply enable it there even before preloads.

I can get some numbers tomorrow for loading nested blocks in `loadBlock`.

Reply:
> We should work with consonant to produce a design to reserve space and place this element consistently within the marquee so that we can avoid waiting for the element to download.

i may be off topic @mokimo @yesil but am i right considering that reserving space will not save us from harming LCP as once mnemonics/countdown arrives, there will be repaint?

Reply:
am trying to understand what would be the pattern for such items, so we can call other usages anti patterns :)

Reply:
> i may be off topic @mokimo @yesil but am i right considering that reserving space will not save us from harming LCP as once mnemonics/countdown arrives, there will be repaint?

@npeltier the way it works is
1. Load LCP block (marquee)
2. Marquee's _image_ is identified as LCP
3. Whatever else loads (countdown, globalnav, etc) are irrelevant. *

*Bigger elements might count as LCP (georouting image) if they are bigger than the marquees image
*Google's LCP algo to detect the LCP stops as a user interacts with the page

Reply:
Delaying the countdown = less requests, bytes and code to be loaded = faster LCP
Reserving the space = avoiding CLS when it eventually loaded (much like the gnav)

Reply:
There is a discussion on nested blocks / fragments etc. here https://github.com/orgs/adobecom/discussions/2819#discussioncomment-10534266 - I'm happy to be challenged in that regard, as we currently still try to understand the concrete numbers // disadvantages of fragments/nested blocks (conceptually equal) within the LCP

Reply:
but isn't the whole container (marquee) re-painted as soon as the timer is rendered? And thus delaying timer rendering would make LCP even bigger?

Reply:
beside creating a CDT-Marquee which does not look great (or does it? temp/bin components folder?)
Current PR looks like the only solution to me together probably with preload/prefetches